src/utils/customPropTypes.js

```diff
@@ -1,11 +1,36 @@
+import {Children} from 'react';
 import _ from 'lodash';
 
 const customPropTypes = {
+  /*
+   * Ensures children are of a set of types
+   * Similar to `oneOfType` built-in validator
+   * @param {Array<string>} allowedTypes Collection of allowed Stardust component types
+   * @returns {Array} containing children of the specified type
+   */
+  childrenOfTypesOnly: (allowedTypes) => {
+    return (props, propName, componentName) => {
+      const {children} = props;
+      const disallowed = Children.map(children, child => {
+        return _.includes(allowedTypes, _.get(child, 'type._meta.name')) ? null : child;
+      });
+      if (disallowed && disallowed.length !== 0) {
+        return new Error(
+          `\`${componentName}\` should only have children of type \`${allowedTypes}\`.`);
```

Comment:
Looks like the closing paren should be on the next line aligned with the `return`, or the entire error on one line.